lib/rules/no-global-jquery.js

```diff
@@ -28,96 +26,13 @@ module.exports = {
-
     return {
-      ImportDeclaration(node) {
-        const validImportAlias = new Set(['ember', 'jquery']);
-        // Track if the 'jquery' and/or 'ember' module (if an application
-        // is not using the new modules import syntax) is being imported.
-        // Will exclude non-ember or non-jquery imports such as:
-        //    - `import Foo from 'bar';`
-        //    - `import { readOnly } from '@ember/object/computed';`
-        if (validImportAlias.has(node.source.value)) {
-          const isJqueryImport = node.source.value === 'jquery';
-          if (isJqueryImport) {
-            hasJqueryImport = true;
-            importAliasName = node.source.value;
-          } else {
-            const emberImport = ember.getEmberImportAliasName(node);
-            if (emberImport) {
-              hasEmberImport = true;
-              importAliasName = emberImport;
-            }
-          }
-        }
-      },
-
-      VariableDeclarator(node) {
-        // If an application is using an older version of Ember, not using
-        // the new modules import syntax, then lets locate a destructured
-        // jQuery assignment.
-        if (hasEmberImport) {
-          // Let's verify that we are dealing with an Ember MemberExpression
-          // only to test against.
-          if (
-            node.init &&
-            types.isMemberExpression(node.init) &&
-            isEmberMemberExpression(node.init)
-          ) {
-            const isJQueryVariable = node.init.property.name === '$';
-
-            if (isJQueryVariable) {
-              // Assignment of type const $ = Ember.$;
-              destructuredAssignment = node.id.name;
-            }
-          } else {
-            // If we are not dealing with an Ember identifier, there is no need
-            // to perform the below checks/run the below code.
-            if (node.init && !isEmberIdentifier(node.init)) {
-              return;
-            }
-
-            if (!node.id.properties || !isNestedJQueryAssignment(node.id.properties)) {
-              return;
-            }
-            // Assignment of type const { $: foo } = Ember;
-            // It will grab/return "foo".
-            destructuredAssignment = utils
-              .collectObjectPatternBindings(node, {
-                [importAliasName]: ['$'],
-              })
-              .pop();
-          }
-        }
-      },
-
-      CallExpression(node) {
-        // In the event in which the jQuery module is being imported
-        // using the new modules import syntax do not report to ESLint
-        if (hasJqueryImport) {
-          return;
-        }
-
-        if (utils.isGlobalCallExpression(node, destructuredAssignment, ALIASES)) {
-          context.report(node, ERROR_MESSAGE);
-        }
-      },
+      /**
+       * Report usages of global jQuery
+       *
+       * eg; $(body) and $.post()
+       */
+      'Program:exit': reportGlobals(context, ALIASES, ERROR_MESSAGE),
```

Comment:
In #1066, I went ahead and refactored the `no-global-jquery` rule to use the [ReferenceTracker](https://eslint-utils.mysticatea.dev/api/scope-utils.html#referencetracker-class) class which provides a straightforward way for detecting global function calls and which we already use in several rules. You can rebase to pick up that change, and ideally you could use ReferenceTracker in `no-jquery` too for the same purpose.